### PR TITLE
Contact Management 2 (#585)

### LINKED ISSUE #585: Contact Management
at the moment it is not clear how we should delete or move a contact between groups.

we also have no way to create/remove/edit groups

### PR DESCRIPTION
**Pull request by flexsurfer**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)
fixes #585 #827

### Summary:
[comment]: # (Summarise the problem and how the pull request solves it)
Implemented groups reordering
Fixed styles (code review https://github.com/status-im/status-react/pull/790)

### Steps to test:
- Open Status
- Navigate to Contacts Tab

[comment]: # (PRs will only be accepted if squashed into single commit.)
status: ready

### CODE REVIEW

**Comment by rasom** [comment]
Reordering:

1. Intersection of items on selection is a bit weird, i suppose it is better to keep the previous item in its place before user starts moving selected item, but maybe that's how `react-native-sortable-listview` works

ord

2. Label is missed, zeplin:
screen shot 2017-02-26 at 11 51 35

Android device:
screenshot_2017-02-26-11-51-11

**Comment by flexsurfer** [comment]
1 - strange behavior, I'll look, it worked goo
2- yes, we're discussing with @andmironov do we need this or not

**Comment by rasom** [comment]
Context menu position is wrong [Android]:
screenshot_2017-02-26-13-25-19

Zeplin:
screen shot 2017-02-26 at 14 19 06

**Comment by flexsurfer** [comment]
Context menu position is wrong - @andmironov this is a common behavior for the popup menu on android

**Comment by rasom** [comment]
"Contacts" label moves a bit on switching between screens:
screenshot_2017-02-26-14-28-56

cntcts

**Comment by rasom** [comment]
Grey block at the bottom of group details screen
screenshot_2017-02-26-14-41-27

**Comment by rasom** [comment]
screenshot_2017-02-26-14-54-36

1. Medium font, should be regular
2. Bottom line has wrong styles

screen shot 2017-02-26 at 14 59 06

**Comment by rasom** [comment]
Oh, also "Favourites" is underlined on the previous screen, this is wrong.

**Comment by flexsurfer** [comment]
Grey block at the bottom of group details screen - this is because https://github.com/status-im/status-react/issues/635 not fixed

**Comment by rasom** [comment]
But swiper is not used on that screen

**Comment by flexsurfer** [comment]
you're right

**Comment by rasom** [comment]
three-dots icon is blurred [iOS]
screen shot 2017-02-27 at 09 16 32

**Comment by rasom** [comment]
screen shot 2017-02-27 at 09 59 24
Font styles are wrong, dapp badges are still missed everywhere

**Comment by rasom** [comment]
screen shot 2017-02-27 at 10 04 08
The last separator is missed

**Comment by rasom** [comment]
Also, if you will look at the last screen, it seems that each item has 63pt height and 1pt for separator when on simulator each item has 64pt height, that's why screens look a bit differently

**Comment by rasom** [comment]
screen shot 2017-02-27 at 10 18 55
extra separator [iOS]

File: src/status_im/components/sortable_list_view.cljs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  (:require [reagent.core :as r]
+            [status-im.components.react :refer [view
+                                                touchable-highlight
+                                                list-item]]))
+
+(def sortable-list-view-class (r/adapt-react-class (js/require "react-native-sortable-listview")))
+
+(defn sortable-list-view [{:keys [onRowMoved renderRow] :as props}]
+  [sortable-list-view-class
+   (assoc props :onRowMoved #(onRowMoved (js->clj % :keywordize-keys true))
```
Comment (rasom) [inline src/status_im/components/sortable_list_view.cljs]: Let's use `lisp-case` here for `:on-row-moved` and `render-row`

File: src/status_im/components/toolbar_new/styles.cljs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  {:backgroundColor (or background-color toolbar-background1)
+   :elevation       2})
+
+(def toolbar
+  (merge {:flex-direction :row}
+         (get-in p/platform-specific [:component-styles :toolbar-new])))
+
+(defn toolbar-nav-actions-container [actions]
+  (let [center? (get-in p/platform-specific [:component-styles :toolbar-title-center?])]
+    (merge {:flex-direction "row"}
```
Comment (rasom) [inline src/status_im/components/toolbar_new/styles.cljs]: Let's use keyword `:row` everywhere instead "row"

File: src/status_im/components/toolbar_new/styles.cljs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+           (when center?
+             {:width          (when (and actions (> (count actions) 0))
```
Comment (rasom) [inline src/status_im/components/toolbar_new/styles.cljs]: `(pos? (count actions))`

File: src/status_im/components/toolbar_new/view.cljs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+                    [view st/toolbar-action
+                     [image action-image]]])
+             {:key (str "action-" action-image)}))
+         custom-action)]]
+     [sync-state-gradient-view]
+     [view st/toolbar-border-container
+      [view st/toolbar-border]]]))
+
+(defn- toolbar-search-submit [on-search-submit]
+  (let [text @(subscribe [:get-in [:toolbar-search :text]])]
```
Comment (rasom) [inline src/status_im/components/toolbar_new/view.cljs]: Is there a way to not use subscription here? It doesn't look like the best solution

File: src/status_im/components/toolbar_new/view.cljs [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+                                   search-key
+                                   nav-action
+                                   actions
+                                   style
+                                   on-search-submit]
+                            :as   opts}]
+  (let [toggle-search-fn #(do
+                            (dispatch [:set-in [:toolbar-search :show] %])
+                            (dispatch [:set-in [:toolbar-search :text] ""]))
+        actions          (if-not show-search?
```
Comment (rasom) [inline src/status_im/components/toolbar_new/view.cljs]: `(when-not show-search?`

File: src/status_im/components/toolbar_new/view.cljs (hunk 4 above)
Comment (flexsurfer) [inline src/status_im/components/toolbar_new/view.cljs]: this is copy/paste from the old toolbar, i'll clean unused code

**Comment by flexsurfer** [comment]
- [ ] Reordering: Intersection of items on selection is a bit weird

- [ ] Reordering: Label is missed

- [ ] Context menu position is wrong [Android]

- [ ] "Contacts" label moves a bit on switching between screens

- [ ] Grey block at the bottom of group details screen

- [ ] Medium font, should be regular

- [ ] Bottom line has wrong styles

- [ ] "Favourites" is underlined, this is wrong

- [ ] three-dots icon is blurred [iOS]

- [ ] contacts [iOS] Font styles are wrong 

- [ ] dapp badges are still missed everywher

- [ ] The last separator is missed [iOS]

- [ ] it seems that each item in the list has 63pt height and 1pt for separator when on simulator each item has 64pt height, that's why screens look a bit differently

- [ ] extra separator [iOS]

- [ ] code review

**Comment by rasom** [comment]
screen shot 2017-02-27 at 13 45 40
Password's bottom line has a wrong color. Also "back" icon has different sizes, but i'm not sure that this is a big problem.

**Comment by rasom** [comment]
>  Bottom line has wrong styles
can't reproduce

I can reproduce this on both genymotion and Android real device

**Comment by rasom** [comment]
Warning on Reordering screen
screen shot 2017-02-27 at 21 38 23

**Comment by rasom** [comment]
Great work @flexsurfer, application looks much better now :)

One more thing, wrong font styles here:
screen shot 2017-02-27 at 22 03 45

**Comment by flexsurfer** [comment]
thank you @rasom for your review with scrupulous attention to details!

**Comment by flexsurfer** [comment]
- [x] Warning on Reordering screen

- [ ] wrong font styles in context menu

**Comment by flexsurfer** [comment]
- reworked contect menu items for android (delete contact is red)

- reworked text input bottom line on android (removed padding)

**Comment by flexsurfer** [comment]
- implemented search input clearing

**Comment by flexsurfer** [comment]
- [ ] dapp badges are still missed everywhere

**Comment by flexsurfer** [comment]
- implemented custom title for a toolbar with search (New group N ( N should be different color) )

**Pull request merged**